core/dbt/contracts/graph/nodes.py

```diff
@@ -99,6 +99,49 @@ def add_macro(self, value: str):
             self.macros.append(value)
 
 
+@dataclass
+class InjectedCTE(dbtClassMixin, Replaceable):
+    id: str
+    sql: str
+
+
+@dataclass
+class CompiledNode:
+    compiled: bool = False
+    compiled_code: Optional[str] = None
+    extra_ctes_injected: bool = False
+    extra_ctes: List[InjectedCTE] = field(default_factory=list)
+    relation_name: Optional[str] = None
```

Comment:
It feels weird and incorrect that `relation_name` is only available [during compilation](https://github.com/dbt-labs/dbt-core/blob/16f529e1d4e067bdbb6a659a622bead442f24b4e/core/dbt/compilation.py#L399), rather than after parsing. I think the reason had been, we need to know [the adapter's `quoting` config](https://github.com/dbt-labs/dbt-core/blob/16f529e1d4e067bdbb6a659a622bead442f24b4e/core/dbt/compilation.py#L202-L204) — but I'm pretty sure we'd be able to know that at parse time too, right?

Reply:
I think so. I agree that it would be nice to have this available at parse time. I think "relation_name" was added by an external contributor, so I'm guessing it was at compile time because it was easier to find a place to put it.

Reply:
We do set "relation_name" for SourceDefinition at parse time.

Reply:
I'll look at setting relation_name in the right place in the pull request to implement the nodes in protobuf messages.

Reply:
I think you're right! Sounds good.